File: Data Structures & Algorithms/count-paths/submission-1.cpp
class Solution {
public:
    vector<vector<int>> vc;
    int dfs(int i , int j,int m,int n ,vector<vector<int>>vc){
        if (i == (m - 1) && j == (n - 1)) {
            return 1;
        }
        if (i>=m || j>=n)
            return 0;
        if(vc[i][j]!=-1)
            return vc[i][j];
        vc[i][j]=dfs(i+1,j,m,n,vc)+dfs(i,j+1,m,n,vc);
        return vc[i][j];
    }
    int uniquePaths(int m, int n) {
        vc.resize(m, vector<int>(n, -1));
       
        return  dfs(0,0,m,n,vc);
    }
};

File: Data Structures & Algorithms/surrounded-regions/submission-1.py
class Solution:

    def solve(self, board: List[List[str]]) -> None:
        n,m=len(board),len(board[0])
        def dfs(i,j):
            if i < 0 or j <0 or i>=n or j>=m or board[i][j]!="O":
                return 
            board[i][j]="W"
            dfs(i+1,j)
            dfs(i,j+1)
            dfs(i-1,j)
            dfs(i,j-1)
        
        for r in range(n):
            if board[r][0] == "O":
                dfs(r, 0)
            if board[r][m - 1] == "O":
                dfs(r, m - 1)

        for c in range(m):
            if board[0][c] == "O":
                dfs(0, c)
            if board[n - 1][c] == "O":
                dfs(n - 1, c)

        for r in range(n):
            for c in range(m):
                if board[r][c] == "O":
                    board[r][c] = "X"
                elif board[r][c] == "W":
                    board[r][c] = "O"
